lib/string.ts

```diff
@@ -24,11 +27,31 @@ export function capitalizeOnlyFirstLetter(str: string) {
 function sanitizeMarkdownTableCell(text: string): string {
   return text
     .replaceAll('|', String.raw`\|`)
-    .replaceAll(new RegExp(EOL, 'gu'), '<br/>');
+    .replaceAll(new RegExp(endOfLine, 'gu'), '<br/>');
 }
 
 export function sanitizeMarkdownTable(
   text: readonly (readonly string[])[],
 ): readonly (readonly string[])[] {
   return text.map((row) => row.map((col) => sanitizeMarkdownTableCell(col)));
 }
+
+// Gets the end of line string while respecting the
+// `.editorconfig` and falling back to `EOL` from `node:os`.
+export function getEndOfLine() {
+  // The passed string is the target file name, relative to process.cwd().
+  // Given that the docs markdown files are generated inside the `docs/rules` folder
+  // This should be the correct path to resolve the `.editorconfig` for these
+  // specific markdown files.
+  const config = editorconfig.parseSync('./docs/rules/markdown.md');
```

Comment:
The path to the rule docs is configurable using [--path-rule-doc](https://github.com/bmish/eslint-doc-generator#configuration-options). For that matter, the path to the rule list(s) is configurable too with `--path-rule-list`.

Can we just lookup the `.editorconfig` for the plugin root instead? And not worry about nested doc folders.

Reply:
Sure, in hindsight, having an `.editorconfig` that has a different end_of_line setting only for specific markdown files in a specific folder is probably a really niche case.